---
title: "vignette_group_11_package"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{group_11_package}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
library(group_11_package)
```


When you purify and sequence DNA, you may want to:
* Transcribe the DNA sequence to RNA.
* Get the codons corresponding to the translated RNA.
* Analyse the frequency of the codons in your peptide.

The group_11_package allows you to perform such analysis and also offers the possibility to create a random DNA sequence to practice with its set of tools.

##Creating a random DNA sequence with `randomDNASequence`
`randomDNASequence` takes the length of the DNA sequence as argument and returns a random selection of nucleotides.

For example, we can opt to create a DNA sequence comprised of 20 nucleotides and assign it to a variable:
```{r}
dnaSequence <- randomDNASequence(20)
dnaSequence
```
##Transcribing our DNA sequence with `DnaToRna`
After creating the random DNA sequence or obtaining it elsewhere, `DnaToRna` takes the variable containing the nucleotides as input and transcribes it to RNA.

```{r}
rnaSequence <- DnaToRna(dnaSequence)
rnaSequence
```
##Translating the RNA sequence into a peptide chain with `translateRNASequence`
The next step is translating the RNA sequence to obtain the polypeptide chain and this is possible with `translateRNASequence`

```{r}
peptideSequence <- translateRNASequence(rnaSequence)
peptideSequence
```





